aws/resource_aws_macie2_classification_job_test.go

```diff
@@ -0,0 +1,530 @@
+			},
+			{
+				ResourceName:      resourceName,
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+func testAccCheckAwsMacie2ClassificationJobExists(resourceName string, macie2Session *macie2.DescribeClassificationJobOutput) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		rs, ok := s.RootModule().Resources[resourceName]
+		if !ok {
+			return fmt.Errorf("not found: %s", resourceName)
+		}
+
+		conn := testAccProvider.Meta().(*AWSClient).macie2conn
+		input := &macie2.DescribeClassificationJobInput{JobId: aws.String(rs.Primary.ID)}
+
+		resp, err := conn.DescribeClassificationJob(input)
+
+		if err != nil {
+			return err
+		}
+
+		if resp == nil {
+			return fmt.Errorf("macie ClassificationJob %q does not exist", rs.Primary.ID)
+		}
+
+		*macie2Session = *resp
+
+		return nil
+	}
+}
+
+func testAccCheckAwsMacie2ClassificationJobDestroy(s *terraform.State) error {
+	conn := testAccProvider.Meta().(*AWSClient).macie2conn
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "aws_macie2_classification_job" {
+			continue
+		}
+		input := &macie2.DescribeClassificationJobInput{JobId: aws.String(rs.Primary.ID)}
+
+		resp, err := conn.DescribeClassificationJob(input)
+
+		if tfawserr.ErrMessageContains(err, macie2.ErrCodeAccessDeniedException, "Macie is not enabled") ||
+			tfawserr.ErrMessageContains(err, macie2.ErrCodeValidationException, "cannot update cancelled job for job") {
+			continue
+		}
+
+		if err != nil {
+			return err
+		}
+
+		if resp != nil && aws.StringValue(resp.JobStatus) != macie2.JobStatusCancelled {
+			return fmt.Errorf("macie ClassificationJob %q still exists", rs.Primary.ID)
+		}
+	}
+
+	return nil
+}
+
+func testAccCheckAwsMacie2ClassificationJobNotRecreated(i, j *macie2.DescribeClassificationJobOutput) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		if !aws.TimeValue(i.CreatedAt).Equal(aws.TimeValue(j.CreatedAt)) {
+			return fmt.Errorf("Macie Classification Job recreated")
+		}
+
+		return nil
+	}
+}
+
+func testAccAwsMacieClassificationJobconfigNameGenerated(bucketName, jobType string) string {
+	return fmt.Sprintf(`
+data "aws_caller_identity" "current" {}
+
+resource "aws_macie2_account" "test" {}
+
+resource "aws_s3_bucket" "test" {
+  bucket = "%s"
+}
+
+resource "aws_macie2_classification_job" "test" {
+  depends_on = [aws_macie2_account.test]
+  job_type   = "%s"
+  s3_job_definition {
+    bucket_definitions {
+      account_id = data.aws_caller_identity.current.account_id
+      buckets    = [aws_s3_bucket.test.bucket]
+    }
+  }
+}
+`, bucketName, jobType)
+}
+
+func testAccAwsMacieClassificationJobconfigNamePrefix(nameBucket, namePrefix, jobType string) string {
+	return fmt.Sprintf(`
+data "aws_caller_identity" "current" {}
+
+resource "aws_macie2_account" "test" {}
+
+resource "aws_s3_bucket" "test" {
+  bucket = %q
+}
+
+resource "aws_macie2_classification_job" "test" {
+  name_prefix = %[2]q
+  job_type    = %q
```

Comment:
We shouldn't mix indexed and non-indexed format verbs. They should all be indexed.

```suggestion
resource "aws_s3_bucket" "test" {
  bucket = %[1]q
}

resource "aws_macie2_classification_job" "test" {
  name_prefix = %[2]q
  job_type    = %[3]q
```